### PR TITLE
Attention dropout (0.1) cross-dataset: TF/TFP/AF/DM

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The current model uses **`dropout=0.0`** in every `nn.MultiheadAttention` call (ANPSurfaceDecoder and any attention blocks in ReferenceTransolver). No attention dropout has ever been tried. Attention dropout is a well-established regularizer for transformer-style models — it forces attention heads to learn redundant, distributed representations and prevents over-reliance on a small set of tokens.

In CFD surrogate models, the surface points form a highly structured graph where nearby points are strongly correlated. Without attention dropout, the model may overfit to dominant local patterns in the training meshes rather than learning geometry-invariant pressure distributions. A small attention dropout (0.1) adds negligible compute overhead while potentially acting as a strong regularizer across all 4 datasets.

Reference: Vaswani et al. (2017) original Transformer uses p=0.1; ViT (Dosovitskiy 2020) ablates dropout in self-attention heads extensively.

## Instructions

In `target/icml2026/train.py` (or the model construction code), find every instantiation of `nn.MultiheadAttention` and add `dropout=0.1`. Specifically:

1. In `target/icml2026/core/architectures/senpai_transolver.py`, in `ANPSurfaceDecoder.__init__`, change:
   ```python
   self.cross_attn = nn.MultiheadAttention(
       embed_dim=self.attn_dim,
       num_heads=num_heads,
       dropout=0.0,
       batch_first=True,
   )
   ```
   to:
   ```python
   self.cross_attn = nn.MultiheadAttention(
       embed_dim=self.attn_dim,
       num_heads=num_heads,
       dropout=0.1,
       batch_first=True,
   )
   ```

2. Search for any other `nn.MultiheadAttention` or `dropout=0.0` in `ReferenceTransolver` and related files and apply `dropout=0.1` consistently.

3. Run all 4 datasets with the standard cross-dataset recipe. Use `--wandb_group attn-dropout-0p1` for all runs.

**Run order (sequential on the 8 GPUs — assign 2 GPUs per run):**

### TandemFoil (EMA, Lion):
```bash
cd target/icml2026 && python train.py \
  --dataset tandemfoil \
  --optimizer lion --lr 1.25e-4 --cosine-t-max 10 --grad-clip 1.0 --weight-decay 1e-2 \
  --ema-decay 0.999 \
  --model-slices 64 --model-layers 3 --model-hidden-dim 192 --model-heads 3 \
  --enable-fourier --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only \
  --asinh-pressure --residual-prediction --enable-pressure-prior-addition \
  --epochs 999 \
  --wandb_group attn-dropout-0p1
```

### TandemFoil Paper (EMA):
```bash
cd target/icml2026 && python train.py \
  --dataset tandemfoil_paper \
  --optimizer lion --lr 1.25e-4 --cosine-t-max 10 --grad-clip 1.0 --weight-decay 1e-2 \
  --ema-decay 0.999 \
  --model-slices 64 --model-layers 3 --model-hidden-dim 192 --model-heads 3 \
  --enable-fourier --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only \
  --asinh-pressure --residual-prediction --enable-pressure-prior-addition \
  --epochs 999 \
  --wandb_group attn-dropout-0p1
```

### AirfRANS (EMA, AdamW):
```bash
cd target/icml2026 && python train.py \
  --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 6e-4 --cosine-t-max 10 --grad-clip 1.0 \
  --ema-decay 0.999 \
  --enable-fourier \
  --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 \
  --wandb_group attn-dropout-0p1
```

### DrivAerML (no-EMA, AdamW — mandatory):
```bash
cd target/icml2026 && SENPAI_MAX_EPOCHS=9999 python train.py \
  --dataset drivaerml \
  --optimizer adamw --lr 5e-4 --cosine-t-max 30 --no-use-ema \
  --enable-fourier \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --epochs 999 \
  --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --wandb_group attn-dropout-0p1
```

Note: The `dropout` argument for `nn.MultiheadAttention` only applies during `.train()` mode — val/test inference is unaffected.

### TandemFoil
- **val_primary/surface_pressure_mae:** 26.134 (EMA recipe, PR #2899)
- **W&B run:** nrn0q3ct

### TandemFoil Paper
- **val_primary/field_mse:** No radford baseline established yet — report whatever you get.

### AirfRANS
- **val_primary/surface_mse:** 0.000598 (PR #2906, no-EMA + no-WD, 360-min)
- **W&B run:** d7a0z1hk
- Note: Using EMA for this run — report best achieved val_primary/surface_mse.

### DrivAerML
- **val_primary/surface_rel_l2_pct:** 3.997% (PR #2898, 467 epochs)
- **W&B run:** bht6h42t
- **External target:** <3.71% (AB-UPT)

## Expected outcome

If attention dropout helps, expect ~2-5% improvement on surface metrics. Most likely to help on DrivAerML where overfitting to training meshes is plausible given the smaller dataset.

### CODE REVIEW

**Comment by morganmcg1** [comment]
Closing to narrow the queue for the final ICML benchmark sprint. Current priority is benchmark-specific optimization and benchmark-faithful reporting for DrivAerML, TandemFoil paper benchmark, AirfRANS full Surf/Vol, and the Tandem parity anchor. If this idea still matters later, we can reopen it in a benchmark-specific form.
